test-e2e.sh

```diff
@@ -81,6 +81,22 @@ kubectl version
 # Don't build/test the prow image as it requires Docker
 EXCLUDED_TARGETS="-//images/gcloud-bazel:gcloud_install -//images/gcloud-bazel:gcloud_push"
 
+# Create a unique namespace for this job using the repo name and the build id
+export E2E_NAMESPACE="${BUILD_ID:-0}"
+
+(cat <<EOF
+{
+  "kind": "Namespace",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "${E2E_NAMESPACE}",
+    "labels": {
+    }
+  }
+}
+EOF
```

Comment:
First of all, are you sure `${E2E_NAMESPACE}` is set to be the correct value? Could you use `echo` for debugging and make sure it exists? I am asking because I suspect this script is not started by bazel so the variable is empty. 

Secondly, instead of constructing a JSON by hand, I would simply do 

``` bash
# Before test
kubectl create namespace "${E2E_NAMESPACE}"

# After test
kubectl delete namespace "${E2E_NAMESPACE}"
```

Reply:
Yes I have a "export E2E_NAMESPACE="build-${BUILD_ID:-0}" right above